src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -0,0 +1,25 @@
+package io.treeverse.clients
+
+import org.json4s._
+import org.json4s.native.JsonMethods._
+import scalaj.http.Http
+
+class ApiClient(apiUrl : String, accessKey : String, secretKey : String) {
+  def getRepositoryStorageNamespace(repoName: String): String = {
```

Comment:
I think this should be cached.  Even just in-process, with a long (>2m?) TTL.

Not blocking this PR, of course.